packages/material/src/complex/MaterialArrayControlRenderer.tsx

```diff
@@ -81,33 +80,18 @@ export class MaterialArrayControlRenderer extends RendererComponent<TableControl
     );
   }
 
-  private select = (_event, child) => {
-    const selected = this.state.selected.filter(s => this.props.data.indexOf(s) !== -1);
-    const selectedIndex: number = selected.indexOf(child);
-    let newSelected = [];
-
-    if (selectedIndex === -1) {
-      newSelected = newSelected.concat(selected, child);
-    } else if (selectedIndex === 0) {
-      newSelected = newSelected.concat(selected.slice(1));
-    } else if (selectedIndex === selected.length - 1) {
-      newSelected = newSelected.concat(selected.slice(0, -1));
-    } else if (selectedIndex > 0) {
-      newSelected = newSelected.concat(
-        selected.slice(0, selectedIndex),
-        selected.slice(selectedIndex + 1),
-      );
-    }
+  private select = (_event, index) => {
+    const copy = this.state.selected.slice();
```

Comment:
Does this fix the select issue described in  #894 ?

Reply:
Yes, the whole PR was actually meant to fix #894. I forgot to mention that.